specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/stable/2021-03-07/CommunicationIdentity.json

```diff
@@ -0,0 +1,360 @@
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "description": "Identifier of the identity to issue token for.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Requesting scopes for the new token.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CommunicationIdentityAccessTokenRequest"
+            }
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CommunicationIdentityAccessToken"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Issue an access token": {
+            "$ref": "./examples/IssueAccessToken.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CommunicationIdentityTokenScope": {
+      "description": "List of scopes for an access token.",
+      "enum": [
+        "chat",
+        "voip"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "CommunicationIdentityTokenScope",
+        "modelAsString": true
+      }
+    },
+    "CommunicationIdentityCreateRequest": {
+      "type": "object",
+      "properties": {
+        "createTokenWithScopes": {
+          "description": "Also create access token for the created identity.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationIdentityTokenScope"
+          }
+        }
+      }
+    },
+    "CommunicationError": {
+      "description": "The Communication Services error.",
+      "required": [
+        "code",
+        "message"
+      ],
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "The error code.",
+          "type": "string"
+        },
+        "message": {
+          "description": "The error message.",
+          "type": "string"
+        },
+        "target": {
+          "description": "The error target.",
+          "type": "string",
+          "readOnly": true
+        },
+        "details": {
+          "description": "Further details about specific errors that led to this error.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationError"
+          },
+          "readOnly": true
+        },
+        "innerError": {
+          "$ref": "#/definitions/CommunicationError"
+        }
+      },
+      "x-nullable": true
```

Comment:
Can you comment on why this is necessary?  When does the service return a JSON `null` for the error?

Reply:
Whenever we have a successful response, the "error" is null.
it's the "error" prop in the response, not the response itself that is nullable.

Reply:
@DominikMe  do you want to provide more information?

Reply:
The `x-nullable` extension indicates that the REST endpoint can send/receive an explicit JSON `null` for the specified type on the wire.  Presumably your success/error responses are mutually exclusive yes (looking at the provided examples, the answer is yes)?

Please see [here](https://github.com/Azure/autorest/tree/master/docs/extensions#x-nullable) for more info and some examples.

Reply:
@jhendrixMSFT I think I had copied `x-nullable` over from KeyVault's latest Error definition here:

https://github.com/Azure/azure-rest-api-specs/blob/a35bddecf81df1a1a3030dbaee0b73c777aeb0db/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/common.json#L70-L71

Looking at the guidelines, they say 
> The error response MUST be a single JSON object. This object MUST have a name/value pair named "error." The value MUST be a JSON object.

So I agree that we should remove `x-nullable` and always set it with code/message.

Reply:
Is it something blocking to approve this PR? Do we need to remove it now?

Reply:
If it's not required yes please remove it as it has impact on codegen.

Reply:
removed